qiskit/chemistry/algorithms/minimum_eigen_solvers/oovqe.py

```diff
@@ -0,0 +1,604 @@
+                self.var_form._bounds = self.var_form_bounds
+                vqresult_wavefun = self.find_minimum(
+                    initial_point=self.initial_point[:self.var_form_num_parameters],
+                    var_form=self.var_form,
+                    cost_fn=self._energy_evaluation,
+                    optimizer=self.optimizer)
+                self.initial_point[:self.var_form_num_parameters] = vqresult_wavefun.optimal_point
+
+                # optimize orbitals
+                self.var_form._bounds = self.bound_oo
+                self.var_form._num_parameters = self.orbital_rotation.num_parameters
+                self.fixed_wavefunction_params = vqresult_wavefun.optimal_point
+                vqresult = self.find_minimum(
+                    initial_point=self.initial_point[self.var_form_num_parameters:],
+                    var_form=self.var_form,
+                    cost_fn=self._energy_evaluation_oo,
+                    optimizer=self.optimizer)
+                self.initial_point[self.var_form_num_parameters:] = vqresult.optimal_point
+        else:
+            # simultaneous method (ansatz and orbitals are optimized at the same time)
+            self.var_form._bounds = self.bounds
+            self.var_form._num_parameters = len(self.bounds)
+
+            vqresult = self.find_minimum(initial_point=self.initial_point,
+                                         var_form=self.var_form,
+                                         cost_fn=self._energy_evaluation_oo,
+                                         optimizer=self.optimizer)
+
+        # write original number of parameters to avoid errors due to parameter number mismatch
+        self.var_form._num_parameters = self.var_form_num_parameters
+
+        # save the results
+        self._ret = {}
+        self._ret['num_optimizer_evals'] = vqresult.optimizer_evals
+        self._ret['min_val'] = vqresult.optimal_value
+        if self.iterative_oo:
+            self._ret['opt_params'] = self.initial_point[self.var_form_num_parameters:]
+            self._ret['opt_params_orbitals'] = self.initial_point[:self.var_form_num_parameters]
+        else:
+            self._ret['opt_params'] = vqresult.optimal_point[:self.var_form_num_parameters]
+            self._ret['opt_params_orbitals'] = vqresult.optimal_point[self.var_form_num_parameters:]
+
+        self._ret['eval_time'] = vqresult.optimizer_time
+        self._ret['opt_psarams_dict'] = vqresult.optimal_parameters
+
+        if self._ret['num_optimizer_evals'] is not None and \
+                self._eval_count >= self._ret['num_optimizer_evals']:
+            self._eval_count = self._ret['num_optimizer_evals']
+        self._eval_time = self._ret['eval_time']
+        logger.info('Optimization complete in %s seconds.\nFound opt_params %s in %s evals',
+                    self._eval_time, self._ret['opt_params'], self._eval_count)
+        self._ret['eval_count'] = self._eval_count
```

Comment:
When the optimization is looped over in the iterative oo shouldn't  eval_time be summed through those loops so the final time reflects the true total over all the evaluations? Since eval_count is incremented via energy evaluation I think this one takes care of the complete total by itself.

Reply:
Introduced `total_time` to collect `optimizer_time`. Not sure why you mentioned `eval_count`.